### DISCUSSION TITLE
How to add a menu item for a page in Pages

### DISCUSSION THREAD

**Discussion by raramuridesign** [joomlatools/joomlatools-pages#758] [Q&A] [answered]
There is a bit of info that when we create menus, we should use the menu manager in Joomla to select pages and then link to this for the menu, however I have not seen this option in the component.

My questions
1. Is it possible to link to a HOMEPAGE file? Joomla keeps giving an error saying it requires a component 
2. The reference here https://github.com/joomlatools/joomlatools-pages/wiki/Quick-Start to menus
The Menu Item Type should be Pages > Default.
Does not exist

If you can advise what we are doing wrong here. Thanks.
Matthew

**Comment by johanjanssens** [comment, marked as answer]
Happy to help. The documentation is a bit outdated, I haven't had the change to update it. I [did so now.](https://github.com/joomlatools/joomlatools-pages/wiki/Quick-Start)  

Basically you can use any kind of menu item, it doesn't matter. If you create a page for the same url path, Pages will override that page, it will render your page instead of letting Joomla render the menu item.  

### System Links > URL

If you don't like to link to a component you can use `System Links > URL` menu type. The URL should be the exact path of the page relative to your `/joomlatools-pages` directory. 

For example, if you are adding a page '/joomlatools-pages/foo/bar.html.php', the internal URL should be `/foo/bar`. The alias can be anything Joomla doesn't use it to generate the url for the menu item. 

####  Exception: The homepage

For the homepage you cannot use a System Link > URL, Joomla doesn't allow to make an internal url the homepage, so you can use any other kind of menu item. It doesn't really matter at all.

### Menu Item Decoration
 
Decoration means that instead of pages handled the generation of the page, you let Joomla render the component, and you allow Pages to decorate (add content around) the component output. 

This is very similar to a template override, or instead of injecting modules into the menu item, you add additional content using pages partials. 

For more info please read: https://github.com/joomlatools/joomlatools-pages/pull/654 which explains how menu item decoration works with the changes in 0.21.

### Component Decoration

A even more advanced approach to decorate Joomla components is by not creating the menu item at all, but directly specifying the component, view and layout and additional params.

For more info please read: https://github.com/joomlatools/joomlatools-pages/pull/659 which explains component decoration.

_Note:_  I'm still working to allow you to inject menu items into Joomla when using component decoration, this is planned but didn't have time to finish this yet.

**Reply by raramuridesign** [reply]
@johanjanssens I have tried to override the home link, by adding the alias url as a page - but no luck.
Not sure what I am missing...

**Reply by johanjanssens** [reply]
It's not the alias that matters most in Pages, it's the actual path of the url, which defines the path of the page inside your `/joomlatools-pages/pages`directory.

The Joomla home menu item its url is `https://www.example.com` it doesn't have a url path. In Pages you would add a `/joomlatools-pages/pages/index.html.php` file to target it.

This is explained in more detail here: https://github.com/joomlatools/joomlatools-pages/wiki/Folder-Structure#pages
